sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/DecryptorV2.java

```diff
@@ -65,7 +65,7 @@ Flux<ByteBuffer> decrypt(Flux<ByteBuffer> encryptedFlux, EncryptedBlobRange encr
                     return Mono.error(LOGGER.logExceptionAsError(Exceptions.propagate(e)));
                 }
 
-                ByteBuffer decryptedRegion = ByteBuffer.allocate(gcmEncryptionRegionLength);
+                ByteBuffer decryptedRegion = ByteBuffer.allocate((int) authenticatedRegionDataLength);
```

Comment:
It might be worth just casting the `authenticatedRegionDataLength` [here](https://github.com/Azure/azure-sdk-for-java/pull/41685/files#diff-a4036e19a5398a3d8683a2b2e5b5e555dbbcb6a72355f8738dd089e478df2227R48) that way we are fully using integers throughout this implementation and when we go to support longs end to end, it will be less lines we need to change.